cmd/mole/main.go

```diff
@@ -57,6 +75,93 @@ func main() {
 	}
 }
 
+func stopDaemon(app cli.App) error {
+	daemonDir := fmt.Sprintf("%s/%s", instancesDir, app.Stop)
+	pidPathName := fmt.Sprintf("%s/pid", daemonDir)
+	logPathName := fmt.Sprintf("%s/mole.log", daemonDir)
+
+	cntxt := &daemon.Context{
+		PidFileName: pidPathName,
+		PidFilePerm: 0644,
+		LogFileName: logPathName,
+		LogFilePerm: 0640,
+		Umask:       027,
+		Args:        os.Args,
+	}
+
+	d, err := cntxt.Search()
+
+	err = d.Kill()
+	if err != nil {
+		return err
+	}
+	conf, _ := storage.FindByName(app.Stop)
+	var removePath string
+	if conf == nil {
+		removePath = fmt.Sprintf("%s/%s", instancesDir, app.Stop)
+	} else {
+		removePath = fmt.Sprintf("%s/%s/pid", instancesDir, app.Stop)
+	}
+	err = os.RemoveAll(removePath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createInstancesDir() error {
+	_, err := os.Stat(instancesDir)
+	if os.IsNotExist(err) {
+		err = os.MkdirAll(instancesDir, 0755)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func startDaemonProcess(aliasName string) error {
+	cntxt := &daemon.Context{
+		PidFileName: "pid",
+		PidFilePerm: 0644,
+		LogFileName: "mole.log",
+		LogFilePerm: 0640,
+		Umask:       027,
+		Args:        os.Args,
+	}
+	d, err := cntxt.Reborn()
+	if err != nil {
+		return err
+	}
+	if d != nil {
+		daemonDir := fmt.Sprintf("%s/%s", instancesDir, aliasName)
+		pidPathName := fmt.Sprintf("%s/pid", daemonDir)
+		logPathName := fmt.Sprintf("%s/mole.log", daemonDir)
+		if _, err := os.Stat(daemonDir); os.IsNotExist(err) {
+			err := os.Mkdir(daemonDir, 0755)
+			if err != nil {
+				return err
+			}
+		}
+		if _, err := os.Stat(pidPathName); !os.IsNotExist(err) {
+			return errors.New("daemon is already running")
+		}
+		err := os.Rename("pid", pidPathName)
+		if err != nil {
+			return err
+		}
+		err = os.Rename("mole.log", logPathName)
+		if err != nil {
+			return err
+		}
+		log.Printf("mole is running under the alias %s.\nExecute \"mole -stop %s\" if you like to stop it at any time", aliasName, aliasName)
```

Comment:
Multiple lines on a single log message is discouraged.

```suggestion
		 log.Infof("execute \"mole -stop %s\" if you like to stop it at any time", aliasName)
```